model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -318,9 +295,9 @@ def validate_num_shards(
             raise ObjectHasInvalidValueException(
                 f"Num shard {num_shards} must be the same as number of GPUs {gpus} for DeepSpeed."
             )
-    if num_shards > gpus:
+    if num_shards != gpus:
```

Comment:
should we just deprecate the `num_shards` field at this point?

Reply:
yes we should but would prefer not in this PR

Reply:
created https://linear.app/scale-epd/issue/MLI-2315/deprecate-num-shards-field